File: TP3/JAVA/src/test/resources/TestDecorator.java
package test.resources;

import org.junit.jupiter.api.Test;

import main.resources.EtudiantBoursier;
import src.Etudiant;

import static org.junit.jupiter.api.Assertions.*;

public class TestDecorator {

    @Test
    void testEtudiantBoursierDecorator() {
        Etudiant alice = Etudiant("Alice", 20, "E001", "Python");
        EtudiantBoursier boursiere = EtudiantBoursier(alice);
        assertEquals(boursiere.getNom(), "Alice");
    }
}

File: TP4/PYTHON/test/test_decorator.py
from src.codes_initiaux.etudiant import Etudiant
from src.pattern.Decorator.EtudiantBoursier import EtudiantBoursier
from src.pattern.Decorator.EtudiantDelegue import EtudiantDelegue

def test_etudiant_boursier_decorator():
    alice = Etudiant("Alice", 20, "E001", "Python")
    boursiere = EtudiantBoursier(alice)
    assert boursiere.get_nom() == "Alice"
    status = boursiere.afficher_statut()
    assert "boursier" in status.lower()
    assert "exonéré" in status.lower()

def test_etudiant_delegue_decorator():
    bob = Etudiant("Bob", 21, "E002", "Python")
    delegue = EtudiantDelegue(bob)
    
    assert delegue.get_nom() == "Bob"
    action = delegue.organiser_reunion()
    assert "réunion" in action.lower()